python/cuml/cuml/tests/test_kmeans.py

```diff
@@ -171,7 +171,11 @@ def test_weighted_kmeans(nrows, ncols, nclusters, max_weight, random_state):
     sk_kmeans.fit(cp.asnumpy(X), sample_weight=wt)
     sk_score = sk_kmeans.score(cp.asnumpy(X))
 
-    assert cu_score - sk_score <= cluster_std * 1.5
+    if cu_score < sk_score:
+        relative_tolerance = 0.1  # allow 10% difference
+        diff = abs(cu_score - sk_score)
+        avg_score = (abs(cu_score) + abs(sk_score)) / 2.0
+        assert diff / avg_score <= relative_tolerance
```

Comment:
I like the idea of changing these tests from "cuML should only differ from `scikit-learn` by a certain amount" to "cuML should either perform better than `scikit-learn`, the same, or only a little bit worse".